vql/linux/audit.go

```diff
@@ -37,15 +43,54 @@ func (self *streamHandler) outputMultipleMessages(msgs []*auparse.AuditMessage)
 	self.output_chan <- event
 }
 
+type _AuditPluginArgs struct {
+        Rules      []string            `vfilter:"optional,field=rules,doc=List of rules"`
+}
+
 type AuditPlugin struct{}
 
+
 func (self AuditPlugin) Info(scope vfilter.Scope, type_map *vfilter.TypeMap) *vfilter.PluginInfo {
 	return &vfilter.PluginInfo{
 		Name: "audit",
 		Doc:  "Register as an audit daemon in the kernel.",
 	}
 }
 
+func addRules(c *libaudit.AuditClient, rulelst []string, s vfilter.Scope) (error) {
+       for _, line := range rulelst {
+              r, err := flags.Parse(line)
+              if err != nil {
+                     s.Log("Error parsing Rule %v: %s", err)
+                     return fmt.Errorf("Error while adding rules: %w", err)
+               }
+               data, err := rule.Build(r)
+               if err != nil {
+                      s.Log("Error Building Rule: %s", err)
+                      return fmt.Errorf("Error while adding rules: %w", err)
+               }
+               defer c.GetRules()
```

Comment:
Why do you do GetRules() in a deferred context? It simply returns all rules in  binary form i.e not doing any refcounting?

Reply:
This is part of the code that is removed in the subsequent commit